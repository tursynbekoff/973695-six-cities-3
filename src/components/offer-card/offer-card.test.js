import React from "react";
import renderer from "react-test-renderer";
import Card from "./offer-card.jsx";

const index = 5464564;

const offer = {
  price: `60`,
  description: `Wood and stone place`,
  type: `Private room`
};


it(`Should WelcomeScreen render correctly`, () => {
  const onBookmarkClick = jest.fn();

  const tree = renderer
    .create(<Card
      offerCard={offer}
      onBookmarkClick={onBookmarkClick}
      key={index}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
