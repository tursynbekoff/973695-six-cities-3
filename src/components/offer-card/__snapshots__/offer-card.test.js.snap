// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should WelcomeScreen render correctly 1`] = `
<article
  className="cities__place-card place-card"
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="img/apartment-01.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          60
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button"
        onClick={[MockFunction]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height="19"
          width="18"
        >
          <svg
            id="icon-bookmark"
            viewBox="0 0 17 18"
          >
            <path
              d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
            />
          </svg>
        </svg>
        <span
          className="visually-hidden"
        >
          To bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="#"
      >
        Wood and stone place
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      Private room
    </p>
  </div>
</article>
`;
